README.md

```diff
@@ -305,45 +345,64 @@ The Nuxeo Platform replies with an access token:
 
 There are some utility methods on the client to help you with this flow:
 
-__Nuxeo.oauth2.getAuthorizationURL(baseURL, clientId[, params])__
+__Nuxeo.oauth2.getAuthorizationURL(opts)__
 
-Returns the OAuth2 authorization URL for the given `baseURL` and `clientId`.
+Returns the OAuth2 authorization URL for the given `opts` object, containing `baseURL`, `clientId` and optional `clientSecret` and `params`.
 
 ```javascript
-var authorizationURL = Nuxeo.oauth2.getAuthorizationURL('http://localhost:8080/nuxeo', 'my-app', {
-  state: 'xyz',
-  redirect_uri: 'http://localhost:8000/authorize',
+var authorizationURL = Nuxeo.oauth2.getAuthorizationURL({
+  baseURL: 'http://localhost:8080/nuxeo',
+  clientId: 'my-app',
+  clientSecret: 'my-secret', // required if the client defines a secret
+  params: {
+    state: 'xyz',
+    redirect_uri: 'http://localhost:8000/authorize',
+  }
 });
 console.log(authorizationURL); // http://localhost:8080/nuxeo/oauth2/authorize?client_id=my-app&response_type=code&state=xyz&redirect_uri=http://localhost:8000/authorize
 ```
 
-__Nuxeo.oauth2.fetchAccessTokenFromAuthorizationCode(baseURL, clientId, code[, params])__
+__Nuxeo.oauth2.fetchAccessTokenFromAuthorizationCode(opts)__
 
-Fetches an OAuth2 access token for the given `baseURL`, `clientId` and `code`.
+Fetches an OAuth2 access token for the given `opts` object containing `baseURL`, `clientId`, `code` and optional `clientSecret` and `params`.
 
 ```javascript
 var code = ...
-Nuxeo.oauth2.fetchAccessTokenFromAuthorizationCode('http://localhost:8080/nuxeo', 'my-app', code, {
-  redirect_uri: 'http://localhost:8000/authorize',
+Nuxeo.oauth2.fetchAccessTokenFromAuthorizationCode({
+  baseURL: 'http://localhost:8080/nuxeo',
+  clientId: 'my-app',
+  clientSecret: 'my-secret', // required if the client defines a secret
+  code: code,
+  params: {
+    redirect_uri: 'http://localhost:8000/authorize',
+  }
 }).then(function(token) {
   // do something with the access token
   var nuxeo = new Nuxeo({
     auth: {
       method: 'bearerToken',
-      token: token
+      token: token,
+      clientId: 'my-app', // optional OAuth2 client ID to refresh the access token
+      clientSecret: 'my-secret', // required if the client defines a secret
     }
   });
 });
 ```
 
-__Nuxeo.oauth2.fetchAccessTokenFromJWTToken(baseURL, clientId, jwtToken[, params])__
+__Nuxeo.oauth2.fetchAccessTokenFromJWTToken(opts)__
 
-Fetches an OAuth2 access token for the given `baseURL`, `clientId` and `jwtToken`.
+Fetches an OAuth2 access token for the given `opts` object containing `baseURL`, `clientId`, `jwtToken` and optional `clientSecret` and `params`.
```

Comment:
```suggestion
Fetches an OAuth2 access token for the given `opts` object containing `baseURL`, `clientId`, `jwtToken`, and optional `clientSecret` and `params`.
```

Reply:
containing `baseURL`, `clientId`, `jwtToken` optional `clientSecret` (and) `params`